File: src/main/models/SiteTag.ts
/**
 * 站点标签
 */

export default class SiteTag {
  /**
   * 主键
   */
  id: number
  /**
   * 标签来源站点id
   */
  siteId: number
  /**
   * 站点中标签的id
   */
  siteTagId: string
  /**
   * 站点中标签的名称
   */
  siteTagName: string
  /**
   * 上级标签id
   */
  baseSiteTagId: string
  /**
   * 描述
   */
  description: string
  /**
   * 站点标签对应的本地标签id
   */
  localTagId: number
  constructor(siteTag: SiteTag) {
    this.id = siteTag.id
    this.siteId = siteTag.siteId
    this.siteTagId = siteTag.siteTagId
    this.siteTagName = siteTag.siteTagName
    this.baseSiteTagId = siteTag.baseSiteTagId
    this.description = siteTag.description
    this.localTagId = siteTag.localTagId
  }
}

File: src/main-go/internal/model/site_tag.go
package model

// SiteTag 站点标签
type SiteTag struct {
	ID            int64  `gorm:"primaryKey;column:id" json:"id"`
	SiteID        int64  `gorm:"column:site_id;uniqueIndex:idx_site_tag_site_site_tag" json:"siteId"`
	SiteTagID     string `gorm:"column:site_tag_id;uniqueIndex:idx_site_tag_site_site_tag" json:"siteTagId"`
	SiteTagName   string `gorm:"column:site_tag_name" json:"siteTagName"`
	BaseSiteTagID string `gorm:"column:base_site_tag_id" json:"baseSiteTagId"`
	Description   string `gorm:"column:description" json:"description"`
	LocalTagID    int64  `gorm:"column:local_tag_id" json:"localTagId"`
	LastUse       int64  `gorm:"column:last_use" json:"lastUse"`
	CreateTime    int64  `gorm:"column:create_time" json:"createTime"`
	UpdateTime    int64  `gorm:"column:update_time" json:"updateTime"`
}

func (SiteTag) TableName() string {
	return "site_tag"
}

func (e SiteTag) GetID() int64 {
	return e.ID
}

func (e SiteTag) GetCreateTime() int64 {
	return e.CreateTime
}

func (e SiteTag) GetUpdateTime() int64 {
	return e.UpdateTime
}

func (e SiteTag) SetCreateTime(time int64) {
	e.CreateTime = time
}

func (e SiteTag) SetUpdateTime(time int64) {
	e.UpdateTime = time
}
